web/api/query.go

```diff
@@ -48,6 +68,12 @@ func (serv MetricsService) Query(w http.ResponseWriter, r *http.Request) {
 	expr := params.Get("expr")
 	asText := params.Get("asText")
 
+	timestamp, err := parseTimestampOrNow(params.Get("timestamp"))
+	if err != nil {
+		http.Error(w, fmt.Sprintf("invalid query timestamp %s", err), http.StatusBadRequest)
```

Comment:
Hum... but we don't check the `Accept` header at all anyways. Arguably we should? Not sure that matters enough here though.

I also found https://groups.google.com/forum/#!topic/golang-nuts/CRtVzvFQTII, where there's no real agreement, but it seems in general OK to return HTTP errors with non-`text/plain` `Content-Type`, just one has to write their own HTTP error function then, like InfluxDB does:

https://github.com/influxdb/influxdb/blob/master/httpd/handler.go#L600-L613

Ok, then I'll do the same :)